libs/timebar/src/chartsNew/tracks-events.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useMemo } from 'react'
+import cx from 'classnames'
+import { TimelineContextProps, TimelineScale } from '../types'
+import TimelineContext from '../timelineContext'
+import ImmediateContext from '../immediateContext'
+import { DEFAULT_CSS_TRANSITION } from '../constants'
+import { TimebarChartData, TimebarChartItem, TrackEventChunkProps } from './common/types'
+import styles from './tracks-events.module.css'
+import { useFilteredChartData, useClusteredChartData } from './common/hooks'
+import { getTrackY } from './common/utils'
+
+const getTracksEventsWithCoords = (
+  tracksEvents: TimebarChartData,
+  outerScale: TimelineScale,
+  graphHeight: number
+) => {
+  // TODO merge with Tracks' getTracksWithCoords
+  return tracksEvents.map((trackEvents, trackIndex) => {
+    const baseTrackY = getTrackY(tracksEvents.length, trackIndex, graphHeight)
+    const trackItemWithCoords: TimebarChartItem = {
+      ...trackEvents,
+      y: baseTrackY.defaultY,
+      chunks: !trackEvents.chunks
+        ? []
+        : trackEvents.chunks.map((chunk) => {
+            const x = outerScale(chunk.start)
+            const x2 = chunk.end ? outerScale(chunk.end) : x
+            const width = Math.max(1, x2 - x)
+            return {
+              ...chunk,
+              x,
+              width,
+            }
+          }),
+    }
+    return trackItemWithCoords
+  })
+}
+
+const TracksEvents = ({
+  data,
+  useTrackColor,
+}: {
+  data: TimebarChartData
+  useTrackColor: boolean
+}) => {
+  const { immediate } = useContext(ImmediateContext) as any
+  const {
+    outerScale,
+    outerWidth,
+    graphHeight,
+    tooltipContainer,
+    start,
+    end,
+    outerStart,
+    outerEnd,
+  } = useContext(TimelineContext) as TimelineContextProps
```

Comment:
`as any` and `as TimelineContextProps` not needed anymore after the 2point5 merge